src/estimation/calculate_weights.py

```diff
@@ -98,6 +98,8 @@ def calculate_weighting_factor(
 
     for cell_number, cell_group in groupd_by_cell:
 
+        # cell_number += 1 TODO: investigate if this is needed in HDFS
+
```

Comment:
Yes, there's a key error as it's looking for key 0, don't know why.

Does this cell_number += 1 need to be conditional on dev_test == True, or will it work for both?

Reply:
An alternative would be to follow what we do in TMI (though this hasn't been tested in devtest)

`    grp = filtered_df.groupby("imp_class")`
`    class_keys = list(grp.groups.keys())`

`    for var in target_variable_list:`
`        for k in class_keys:`
`            # Get subgroup dataframe`
`            subgrp = grp.get_group(k)`

Reply:
Thanks for the fix @AnneONS